sdk/src/Services/S3/Custom/Util/AmazonS3Uri.cs

```diff
@@ -34,6 +34,10 @@ public class AmazonS3Uri
     {
         private const string EndpointPattern = @"^(.+\.)?s3[.-]([a-z0-9-]+)\.";
 
+        private static Regex endpointRegexMatch;
+
+        private static Regex EndpointRegexMatch => endpointRegexMatch ?? (endpointRegexMatch = new Regex(EndpointPattern, RegexOptions.Compiled));
```

Comment:
Can you remove `const string EndpointPattern`  and just put the string literal inside the Regex's constructor.  This will help cut down field clutter and be a quick sanity check to make sure that regex pattern isn't referenced anywhere else.

Reply:
Will update EvaluateIfSigV4Required with your updated logic and remove the const.